pandas/core/window/rolling.py

```diff
@@ -1262,16 +1274,48 @@ def count(self):
     """
     )
 
-    def apply(self, func, raw=False, args=(), kwargs={}):
-        from pandas import Series
-
+    def apply(
+        self,
+        func,
+        raw=False,
+        args=None,
+        kwargs=None,
+        engine="cython",
+        engine_kwargs=None,
+    ):
+        if args is None:
+            args = ()
+        if kwargs is None:
+            kwargs = {}
         kwargs.pop("_level", None)
         kwargs.pop("floor", None)
         window = self._get_window()
         offset = _offset(window, self.center)
         if not is_bool(raw):
             raise ValueError("raw parameter must be `True` or `False`")
 
+        if engine == "cython":
+            if engine_kwargs is not None:
+                raise ValueError("cython engine does not accept engine_kwargs")
```

Comment:
test coverage for this?

Reply:
Yup here: https://github.com/pandas-dev/pandas/pull/30151/files#diff-ec0ad82c276c1d0caa9df6105adb8cd7R354